manifests/config.pp

```diff
@@ -106,14 +106,47 @@
     }
   }
 
-  if $::osfamily == 'Debian' {
-    file { '/etc/default/rabbitmq-server':
-      ensure  => file,
-      content => template('rabbitmq/default.erb'),
-      mode    => '0644',
-      owner   => '0',
-      group   => '0',
-      notify  => Class['rabbitmq::service'],
+  case $::osfamily {
+    'Debian': {
+      file { '/etc/default/rabbitmq-server':
+        ensure  => file,
+        content => template('rabbitmq/default.erb'),
+        mode    => '0644',
+        owner   => '0',
+        group   => '0',
+        notify  => Class['rabbitmq::service'],
+      }
+    }
+    'RedHat': {
+      if versioncmp($::operatingsystemmajrelease, '7') >= 0 {
+        file { '/etc/systemd/system/rabbitmq-server.service.d':
+          ensure => directory,
+          owner  => '0',
+          group  => '0',
+          mode   => '0755',
+        } ->
+        file { '/etc/systemd/system/rabbitmq-server.service.d/limits.conf':
+          content => template('rabbitmq/rabbitmq-server.service.d/limits.conf'),
+          owner   => '0',
+          group   => '0',
+          mode    => '0644',
+          notify  => Exec['/usr/bin/systemctl daemon-reload'],
+        }
+        exec { '/usr/bin/systemctl daemon-reload':
+          notify      => Class['Rabbitmq::Service'],
```

Comment:
Please change this to use a rabbitmq specific title, so it avoids a potential collision with other (as yet unknown) modules like so:

    exec { 'rabbitmq-systemd-reload': command => '/usr/bin/systemctl daemon-reload', #...


and change all dependencies to use `Exec['rabbitmq-systemd-reload']`